components/brave_rewards/resources/page/reducers/index.ts

```diff
@@ -24,8 +24,9 @@ const mergeReducers = (state: Rewards.State | undefined, action: any) => {
 
   if (!state) {
     state = storage.defaultState
-    storage.save(state)
-  } else if (state !== startingState) {
+  }
```

Comment:
this is where the problem was as we used save directly and then debounced save came after that and override it. So now we use debounce for this one as well